pkg/relayer/proxy/interface.go

```diff
@@ -24,3 +24,15 @@ type RelayerProxy interface {
 	// and its RelayResponse has been signed and successfully sent to the client.
 	ServedRelays() observable.Observable[*types.Relay]
 }
+
+// RelayServer is the interface of the proxy servers provided by the RelayerProxy.
```

Comment:
Wdyt about updating the last line of the `RelayerProxy` comment, from:
```go
// It is responsible for starting and stopping all supported proxies.
```

To:
```go
// It is responsible for starting and stopping all supported RelayServers.
```

This comment references `RelayerProxy`, so I feel having `RelayerProxy` like referencing `RelayServer` adds clarity to the relationship.

Perhaps we could also add a bit of detail here to give the reader an idea about which axes across which `RelayServer` implementations may be distributed (i.e. serialization, transport, etc.)